### PR TITLE
Add stdout redirection feature for Sulong testing

### PR DESCRIPTION
**Pull request by grimmerm**
Sulong's test harness for its testsuites compares the stdout of a reference execution with Sulong's execution.
This commit adds a feature that allows capturing the stdout of the Sulong execution (Java and native prints) and the whitelist feature.

### CODE REVIEW

File: projects/com.oracle.truffle.llvm.pipe.native/src/capture.cpp [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    do {
+    	bytesRead = read(myPipe[R], &(*buf.begin()), BUF_SIZE);
+    	captured += buf;     	
+    } while(bytesRead == BUF_SIZE);
+
+    if (bytesRead > 0) {
+        buf.resize(bytesRead);
+        captured += buf;
+    }
+    if (myPipe[R] > 0)
```
Comment (mrigger) [inline projects/com.oracle.truffle.llvm.pipe.native/src/capture.cpp]: Can we run `clangformat` on this and consistently use opening and closing brackets?

File: projects/com.oracle.truffle.llvm.test/src/com/oracle/truffle/llvm/test/alpha/CaptureOutputTest.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+package com.oracle.truffle.llvm.test.alpha;
+
+import static org.junit.Assert.assertEquals;
+import org.junit.Test;
+import com.oracle.truffle.llvm.pipe.CaptureOutput;
+
+public class CaptureOutputTest {
+
+    @Test
+    public void testOutputCapturing() {
```
Comment (mrigger) [inline projects/com.oracle.truffle.llvm.test/src/com/oracle/truffle/llvm/test/alpha/CaptureOutputTest.java]: Can we run this in the gate?

File: projects/com.oracle.truffle.llvm.pipe.native/src/capture.cpp [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        close(myPipe[R]);
+}
+
+
+JNIEXPORT jstring JNICALL Java_com_oracle_truffle_llvm_pipe_CaptureOutput_getCapture(JNIEnv * env, jclass c) {
+	std::string::size_type idx = captured.find_last_not_of("\r\n");
```
Comment (mrigger) [inline projects/com.oracle.truffle.llvm.pipe.native/src/capture.cpp]: Why `\r\n`? Is this not just applicable for Windows as a line break, while Linux uses `\n`? Can we have a named constant for that?

File: projects/com.oracle.truffle.llvm.pipe.native/Makefile [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+	@mkdir -p bin/
+	@${JAVA_HOME}/bin/javap -sysinfo -classpath ${MX_CLASSPATH} ${NATIVE_CLASSES} | grep "Classfile " | sed -e 's|Classfile |src_gen/native.h: |' >> $@
+
+OBJECTFILES=${OBJECTS:%.o=bin/%.o}
+CFLAGS=-g -fPIC -I${VPATH}/include -Isrc_gen -I${JAVA_HOME}/include -I${JAVA_HOME}/include/darwin $(shell pkg-config)
+LIBS=-g $(shell pkg-config)
+
+bin/${TARGET}: ${OBJECTFILES}
+	@echo LINK $@
+	@g++ -std=gnu++0x -shared ${LIBS} $+ -o $@
```
Comment (mrigger) [inline projects/com.oracle.truffle.llvm.pipe.native/Makefile]: I am not familiar with Makefiles, but should we use `CXX` here to note that we can use any installed C++ compiler? \cc @rschatz

File: projects/com.oracle.truffle.llvm.pipe.native/src/capture.cpp (hunk 1 above)
Comment (grimmerm) [inline projects/com.oracle.truffle.llvm.pipe.native/src/capture.cpp]: yeah, good idea!

File: projects/com.oracle.truffle.llvm.pipe.native/src/capture.cpp (hunk 3 above)
Comment (grimmerm) [inline projects/com.oracle.truffle.llvm.pipe.native/src/capture.cpp]: I'm not sure; \r\n made my tests work.
Can you answer that, @lxp ?

File: projects/com.oracle.truffle.llvm.test/src/com/oracle/truffle/llvm/test/alpha/CaptureOutputTest.java (hunk 2 above)
Comment (grimmerm) [inline projects/com.oracle.truffle.llvm.test/src/com/oracle/truffle/llvm/test/alpha/CaptureOutputTest.java]: Yeah, I've added tests to gate now.

File: projects/com.oracle.truffle.llvm.pipe.native/Makefile (hunk 4 above)
Comment (grimmerm) [inline projects/com.oracle.truffle.llvm.pipe.native/Makefile]: yeah, cxx works fine!

File: mx.sulong/mx_sulong.py [hunk 5]
```diff
@@ -1161,6 +1170,7 @@ def sulongBuild(args=None):
     'argon2' : runTestArgon2,
     'lifetime' : runLifetimeTestCases,
     'main-arg' : runMainArgTestCases,
+    'pipe' : runPipeTestCases,
```
Comment (mrigger) [inline mx.sulong/mx_sulong.py]: Maybe you can register this as a `mx su-tests` suite? Then we do not need an additional mx command for that.

Comment (mrigger) [inline mx.sulong/mx_sulong.py]: Sorry, it is actually already there. I've overlooked that!

File: mx.sulong/mx_sulong.py [hunk 6]
```diff
@@ -139,7 +141,7 @@ def travis1(args=None):
     """executes the first Travis job (Javac build, benchmarks, polyglot, interop, tck, asm, types, and LLVM test cases)"""
     tasks = []
     with Task('BuildJavaWithJavac', tasks) as t:
-        if t: mx.command_function('build')(['-p', '--warning-as-error', '--no-native', '--force-javac'])
```
Comment (mrigger) [inline mx.sulong/mx_sulong.py]: Maybe we can extract the argument list to a global variable or function call?

**Pull request merged**